certbot-nginx/certbot_nginx/tests/configurator_test.py

```diff
@@ -963,16 +963,33 @@ def test_current_file_hash_in_all_hashes(self):
             "Constants.ALL_SSL_OPTIONS_HASHES must be appended"
             " with the sha256 hash of self.config.mod_ssl_conf when it is updated.")
 
-    def test_old_nginx_version_uses_old_config(self):
+    @mock.patch('ssl.OPENSSL_VERSION_INFO', (1, 1, 1))
+    def test_nginx_version_uses_correct_config(self):
         self.config.version = (1, 5, 8)
         self.assertEqual(os.path.basename(self.config.mod_ssl_conf_src),
                          "options-ssl-nginx-old.conf")
         self._call()
         self._assert_current_file()
         self.config.version = (1, 5, 9)
+        self.assertEqual(os.path.basename(self.config.mod_ssl_conf_src),
+                         "options-ssl-nginx-tls12-only.conf")
+        self._call()
+        self._assert_current_file()
+        self.config.version = (1, 13, 0)
         self.assertEqual(os.path.basename(self.config.mod_ssl_conf_src),
                          "options-ssl-nginx.conf")
 
+    def test_correct_config_for_old_openssl(self):
+        self.config.version = (1, 13, 0)
+        with mock.patch('ssl.OPENSSL_VERSION_INFO', (1, 1, 1)):
+            self.assertEqual(os.path.basename(self.config.mod_ssl_conf_src),
+                             "options-ssl-nginx.conf")
+        with mock.patch('ssl.OPENSSL_VERSION_INFO', (1, 1, 0)):
+            self.assertEqual(os.path.basename(self.config.mod_ssl_conf_src),
+                             "options-ssl-nginx-tls12-only.conf")
+
+
```

Comment:
Can we add a test similar to https://github.com/certbot/certbot/blob/2d3f3a042af6c5a4d2fb1a258e27868d34aef5ae/certbot-apache/certbot_apache/tests/configurator_test.py#L1768 which checks that the hashes of all of the current configuration files are in `ALL_SSL_OPTIONS_HASHES`?